frontend/src/metabase/visualizations/lib/settings/graph.ts

```diff
@@ -115,15 +126,19 @@ export const GRAPH_DATA_SETTINGS = {
       )
         ? [null, "graph.max_categories"] // We want to show "graph.max_categories" setting for the breakout dimension (2nd)
         : [];
+
       return {
         options,
         addAnother:
+          addedDimensions &&
           options.length > addedDimensions.length &&
           addedDimensions.length < maxDimensionsSupported &&
           addedDimensions.every(
             dimension => dimension !== undefined && dimension !== null,
           ) &&
-          vizSettings["graph.metrics"].length < 2
+          (vizSettings["graph.metrics"]
+            ? vizSettings["graph.metrics"].length < 2
+            : false)
```

Comment:
I think I've commented somewhere. I think wrapping the whole thing in `Boolean` makes it reads a little better.

```suggestion
          Boolean(
            vizSettings["graph.metrics"] &&
              vizSettings["graph.metrics"].length < 2,
          )
```

This is very easy to understand coming from the previous version of the code as well as we just satisfy the TS by adding the `vizSettings["graph.metrics"]` check and cast it to a boolean.